applications: nrf5340_audio: bt_stream documentation

erikrobstad opened the pull request:
Made the necessary changes to the doc
to fit the adding of bt_stream.

alexsven commented on doc/nrf/releases_and_maturity/software_maturity.rst:
```diff
@@ -148,21 +148,21 @@ The following table indicates the software maturity levels of the support for th
           * BAP unicast, one CIG with two CIS.
```
BAP was removed for the rest, so it should probably be removed here as well?

erikrobstad replied:
The reason why it was not removed here is that we have not implemented CAP for unicast client. But I don't mind if we remove it

greg-fer commented:
Relates to #32

erikrobstad commented on doc/nrf/images/octave_application_structure.svg:
For the four streamctrl files it should be written "One of the following:"

erikrobstad commented on doc/nrf/images/octave_application_structure_gateway.svg:
For the four streamctrl files it should be written "One of the following:"

greg-fer replied:
There are just two files here. Should I include all four?

erikrobstad replied:
No it is correct to only have the two files

erikrobstad commented on doc/nrf/images/octave_application_structure_sync_module.svg:
For the four streamctrl files it should be written "One of the following:"

erikrobstad reviewed (commented):
@greg-fer I made some minor comments

koffes commented on applications/nrf5340_audio/doc/firmware_architecture.rst:
```diff
@@ -141,13 +148,15 @@ Communication between modules is primarily done through Zephyr's :ref:`zephyr:zb
 The application uses the following buses:
 
-  * ``le_audio_chan`` - For handling LE Audio events from :file:`le_audio_cis_headset.c`, :file:`le_audio_bis_headset.c`, :file:`le_audio_cis_gateway.c`, or :file:`le_audio_bis_gateway.c`.
+  * ``le_audio_chan`` - For handling LE Audio events from the Bluetooth stream modules, specifically :file:`unicast_client.c`, :file:`unicast_server.c`, :file:`broadcast_source.c`, and :file:`broadcast_sink.c`.
```
IMO, this is not something we should document:
- It is quite technical (i.e. for developers)
- It is quite well documented in the code
- There is overhead to maintain this doc.

koffes replied:
No action requested from me, just my two cents :)

koffes commented on doc/nrf/releases_and_maturity/software_maturity.rst:
```diff
@@ -123,9 +123,9 @@ The following table indicates the software maturity levels of the support for th
+          * One BIG with two BIS streams.
           * Audio input: USB or I2S (Line in or using Pulse Density Modulation).
-          * Configuration: 16 bit, several bit rates ranging from 24 kbps to 160 kbps.
+          * Configuration: 16 or 32 bit, several bit rates ranging from 32 kbps to 124 kbps.
```
Can you double check CI coverage on this?

koffes replied:
Goes for all 32 bit mentions.

koffes replied:
Not sure 32 bit should be mentioned until we have discussed. Some modules do not support 32 bit.

erikrobstad replied:
Removed 32 bit for now

Commit:
applications: nrf5340_audio: bt_stream documentation

Also updated the changelog for
the whole Bluetooth architecture
rewrite (bt_management, bt_stream,
bt_rendrer, and bt_content_control).

Signed-off-by: Erik Robstad <erik.robstad@nordicsemi.no>

erikrobstad commented:
Added changelog and rebased
